### PR TITLE
Blog backend

### PR DESCRIPTION
**Pull request by TheBlueSummon**
Getting the following error: Cannot read properties of undefined (reading 'reduce')
I added/moved blog functions in src/client/lib/api

### CODE REVIEW

**Comment by ufsasewebmaster** [comment]
You keep making this mistake; you should be branching from the most recent sprint (sprint-5) and merging back into it, not main. I can't do anything with this pr otherwise because of all the potential outdated changes. Will cherry pick 
fb79167 onto a new branch

File: src/client/lib/api.ts
Comment (ufsasewebmaster) [inline src/client/lib/api.ts:1]: Why are we making a lib folder when we already have an api folder

File: src/client/components/ui/textarea.tsx
Comment (ufsasewebmaster) [inline src/client/components/ui/textarea.tsx:1]: https://ui.shadcn.com/docs/components/textarea
Just use this...

File: src/client/lib/types.ts
Comment (ufsasewebmaster) [inline src/client/lib/types.ts:1]: Don't we already have a zod schema?

**Review by ufsasewebmaster — CHANGES_REQUESTED** [review]
??? If you need help, reach out and find what is wrong incrementally as it comes up, not when everything is done

File: src/client/components/ui/textarea.tsx
Comment (ethanniser) [reply src/client/components/ui/textarea.tsx:1]: this is that?

**Pull request merged**